doc/mpls/MPLS_hld.md

```diff
@@ -0,0 +1,510 @@
+ * @param[in] mode Bulk operation error handling mode.
+ * @param[out] object_statuses List of status for every object. Caller needs to
+ * allocate the buffer
+ *
+ * @return #SAI_STATUS_SUCCESS on success when all objects are removed or
+ * #SAI_STATUS_FAILURE when any of the objects fails to remove. When there is
+ * failure, Caller is expected to go through the list of returned statuses to
+ * find out which fails and which succeeds.
+ */
+typedef sai_status_t (*sai_bulk_set_inseg_entry_attribute_fn)(
+        _In_ uint32_t object_count,
+        _In_ const sai_inseg_entry_t *inseg_entry,
+        _In_ const sai_attribute_t *attr_list,
+        _In_ sai_bulk_op_error_mode_t mode,
+        _Out_ sai_status_t *object_statuses);
+
+/**
+ * @brief Bulk get attribute on In Segment entry
+ *
+ * @param[in] object_count Number of objects to set attribute
+ * @param[in] inseg_entry List of objects to set attribute
+ * @param[in] attr_count List of attr_count. Caller passes the number
+ *    of attribute for each object to get
+ * @param[inout] attr_list List of attributes to set on objects, one attribute per object
+ * @param[in] mode Bulk operation error handling mode
+ * @param[out] object_statuses List of status for every object. Caller needs to
+ * allocate the buffer
+ *
+ * @return #SAI_STATUS_SUCCESS on success when all objects are removed or
+ * #SAI_STATUS_FAILURE when any of the objects fails to remove. When there is
+ * failure, Caller is expected to go through the list of returned statuses to
+ * find out which fails and which succeeds.
+ */
+typedef sai_status_t (*sai_bulk_get_inseg_entry_attribute_fn)(
+        _In_ uint32_t object_count,
+        _In_ const sai_inseg_entry_t *inseg_entry,
+        _In_ const uint32_t *attr_count,
+        _Inout_ sai_attribute_t **attr_list,
+        _In_ sai_bulk_op_error_mode_t mode,
+        _Out_ sai_status_t *object_statuses);
+```
+The existing sai_mpls_api_t structure has been expanded to accommodate the new MPLS bulking APIs:
+```
+/**
+ * @brief MPLS methods table retrieved with sai_api_query()
+ */
+typedef struct _sai_mpls_api_t
+{
+     sai_create_inseg_entry_fn                      create_inseg_entry;
+     sai_remove_inseg_entry_fn                      remove_inseg_entry;
+     sai_set_inseg_entry_attribute_fn               set_inseg_entry_attribute;
+     sai_get_inseg_entry_attribute_fn               get_inseg_entry_attribute;
+
++    sai_bulk_create_inseg_entry_fn                 create_inseg_entries;
++    sai_bulk_remove_inseg_entry_fn                 remove_inseg_entries;
++    sai_bulk_set_inseg_entry_attribute_fn          set_inseg_entries_attribute;
++    sai_bulk_get_inseg_entry_attribute_fn          get_inseg_entries_attribute;
+
+} sai_mpls_api_t;
+```
+
+#### Next Hop
+Full details about SAI Next Hop API and attributes are described here:
+https://github.com/opencomputeproject/SAI/blob/master/inc/sainexthop.h
```

Comment:
please have section to discuss what if the sai attribute is not support in vendor SAI, what is the user perceived behavior.

Reply:
There is no new change in this API, so unsupported attributes should not be handled differently as of today.